src/entity/units/priest.ts

```diff
@@ -105,9 +105,10 @@ const unit: UnitSource = {
       return [];
     }
     const numberOfAlliesToRez = unit.isMiniboss ? 3 : 1;
-    return resurrectableUnits(unit, underworld).filter(u => {
-      return Unit.inRange(unit, u)
-    }).slice(0, numberOfAlliesToRez);
+    return resurrectableUnits(unit, underworld)
+      .filter(u => Unit.inRange(unit, u))
+      .sort((a, b) => math.sqrDistance(a, unit) - math.sqrDistance(b, unit))
```

Comment:
Let's extract this to a util function called `sortClosest` that is meant to be passed into .sort since it's used in many places.  The name will give some added context as to why we're using sqrDistance instead of distance.
Usage:
```
.sort(sortCosestTo(unit))
```
Sample implementation:
```
// returns a function to be passed into a .sort() callback
function sortCosestTo(point: Vec2){
return (a,b) => math.sqrDistance(a, point) - math.sqrDistance(b, point));
}
```
We could do the same with Unit.inRange for convenience:
```
// returns a function to be passed into a .filter() callback
function filterInRangeOf(unit){
   return (target) => math.distance(unit, target) <= unit.attackRange;
}
```

Reply:
Good idea with the sortClosestTo function, will do!

We could do the same for the range one, but it kinda feels like a duplicate of the Unit.inRange() function so I'm not sure we should